fix federated credential administration

strehle opened the pull request:
fix administration of federated credentials, e.g. example client client_federated_jwt_trust

uaac client get client_federated_jwt_trust
 -> one entry
uaac client jwt add client_federated_jwt_trust --issuer http://localhost:8080/uaa/oauth/token --subject admin
uaac client jwt delete client_federated_jwt_trust --issuer http://localhost:8080/uaa/oauth/token --subject admin
uaac client get client_federated_jwt_trust
 -> one entry

strehle committed:
fix federated credential administration

adrianhoelzl-sap commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/ClientJwtCredential.java:
```diff
@@ -45,4 +45,21 @@ public static List<ClientJwtCredential> parse(String clientJwtCredentials) {
+        ClientJwtCredential that = (ClientJwtCredential) object;
+        return subject.equals(that.subject) &&
+               issuer.equals(that.issuer) &&
+               (audience != null ? audience.equals(that.audience) : that.audience == null);
```
```suggestion
               Objects.equals(audience, that.audience);
```

strehle replied:
was done by github co pilot

adrianhoelzl-sap commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/ClientJwtChangeRequest.java:
```diff
@@ -125,7 +125,7 @@ public String getChangeValue() {
     public boolean isFederated() {
         return ((changeMode == ADD || changeMode == UPDATE) && issuer != null && subject != null) ||
-                (changeMode == DELETE && (issuer != null || subject != null));
+                (changeMode == DELETE && (issuer != null && subject != null));
```
Can we simplify the whole expression to `return issuer != null && subject != null;`?

adrianhoelzl-sap commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/ClientJwtChangeRequest.java:
```diff
@@ -125,7 +125,7 @@ public String getChangeValue() {
     @JsonIgnore
     public boolean isFederated() {
```
Can we add a comment here that this corresponds to the RFC 7523 approach?

adrianhoelzl-sap commented on model/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/ClientJwtChangeRequest.java:
```diff
@@ -125,7 +125,7 @@ public String getChangeValue() {
+                (changeMode == DELETE && (issuer != null && subject != null));
     }
 
     @JsonIgnore
```
Can we add a comment here that no explicit JWKS URI is required as the issuer is already a URL (that can be used to look up the token keys)?

dependabot[bot] committed:
build(deps): bump joda-time:joda-time from 2.13.1 to 2.14.0 (#3378)

Bumps [joda-time:joda-time](https://github.com/JodaOrg/joda-time) from 2.13.1 to 2.14.0.
- [Release notes](https://github.com/JodaOrg/joda-time/releases)
- [Changelog](https://github.com/JodaOrg/joda-time/blob/main/RELEASE-NOTES.txt)
- [Commits](https://github.com/JodaOrg/joda-time/compare/v2.13.1...v2.14.0)

---
updated-dependencies:
- dependency-name: joda-time:joda-time
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

strehle committed:
Update model/src/main/java/org/cloudfoundry/identity/uaa/oauth/client/ClientJwtCredential.java

Co-authored-by: Adrian Hölzl <102049638+adrianhoelzl-sap@users.noreply.github.com>

strehle committed:
review

strehle committed:
review

strehle committed:
Merge remote-tracking branch 'origin/develop' into fixFedCreds
